README.md

```diff
@@ -984,6 +984,10 @@ in inheritance.
   is a hybrid of `Array`'s intuitive inter-operation facilities and
   `Hash`'s fast lookup.
 
+*  Prefer `map` over `collect`, `detect` over `find`, and `reduce` over
```

Comment:
I'm thinking we can leave out the reasonings behind the style guide so that the guide is less wordy and easier to digest.